src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,257 @@
+                    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After bug is fixed, remove the below if-statement and the existing code should work.
+                    if (changedDocuments.Any(d => d != document.Id) ||
+                        changedAnalyzerConfigDocuments.Any() ||
+                        changedAdditionalDocuments.Any())
+                    {
+                        return Array.Empty<TextDocumentEdit>();
+                    }
+
+                    // Changed documents
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetDocument, solution.GetDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed analyzer config documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetAnalyzerConfigDocument, solution.GetAnalyzerConfigDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed additional documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetAdditionalDocument, solution.GetAdditionalDocument,
+                        cancellationToken).ConfigureAwait(false);
+                }
+
+                return textDocumentEdits.ToArray();
+
+                static async Task GetTextDocumentEdits<T>(
+                    ArrayBuilder<TextDocumentEdit> textDocumentEdits,
+                    ApplyChangesOperation applyChangesOperation,
+                    Solution solution,
+                    IEnumerable<DocumentId> changedDocuments,
+                    Func<DocumentId, T?> getNewDocumentFunc,
+                    Func<DocumentId, T?> getOldDocumentFunc,
+                    CancellationToken cancellationToken)
+                    where T : TextDocument
+                {
+                    foreach (var docId in changedDocuments)
+                    {
+                        var newDoc = getNewDocumentFunc(docId);
+                        var oldDoc = getOldDocumentFunc(docId);
+                        if (oldDoc == null || newDoc == null)
+                        {
+                            continue;
```

Comment:
this shoudl never happen.  it indicates a bug in the caller since we shoudl only be passing in things that changed, so we must have a before/after.